packages/protocol/docs/batch_auction_design.md

```diff
@@ -0,0 +1,162 @@
+    return new_score > old_score * 1.1
+
+
+```
+
+
+
+Bid increments in English auctions serve as an effective strategy to encourage serious bidding, ensure fair competition, reduce on-chain transaction costs, and minimize proof rewards. By requiring new bids to exceed the current winning bid by a specified percentage in score(e.g., 10% higher), trivial bids are filtered out, enabling the auction to quickly reach the lowest bid per gas.
+
+The function shall be virtual to allow customization in derived smart contracts, enabling flexibility and extensibility.
+
+
+#### Internal metrics
+It is essential to maintain various internal metrics to effectively score bids  and facilitate off-chain analysis:
+
+1. **Average proof time**: This metric represents the moving average delay in proof submission for all proofs used in block verification. Proofs that are not utilized in block verification are excluded from this calculation. The average proof time provides insights into the efficiency of the proof submission process, enabling optimization and monitoring of the overall system performance.
+
+2. **Average bid per gas for verified blocks**: This metric quantifies the average bid per unit of gas for blocks that have successfully passed the verification process. It provides valuable information about bidding behavior and the value assigned to gas consumption in successful blocks.
+
+3. **Average bid per gas for all blocks**: This metric calculates the average bid per unit of gas for all blocks, including those that are currently undergoing the auction process. By considering all blocks, this metric offers a comprehensive view of the average bidding behavior and expenditure on gas across the entire system.
+
+4. **Per bidder proof submission success rate**: This metric measures the success rate of proof submissions by individual bidders. Specifically, it evaluates the ratio of proofs submitted by a bidder that were subsequently used for block verification compared to the total number of blocks won through auctions. Proofs submitted to other blocks that the bidder did not win are excluded from this calculation. This metric allows for the assessment of bidder reliability and the effectiveness of their proof submission process.
+
+
+### Bid Period, Proofing Window, and Managing Multiple Auctions
+The bid period for an auction begins with the first bid and lasts for 5 minutes. Blocks can only be proven after the auction has ended entirely. To ensure fairness, particularly for late participants, it is crucial not to end auctions prematurely. It is recommended that an auction concludes when the following conditions are met:
+
+1. The 5-minute bid period has elapsed.
+2. The most recent proposed block is within `min_blocks_to_end_auctions` blocks of the first block in the batch.
+
+Auctions are conducted in increasing order of block batches, and the next batch's auction can start as soon as the previous batch's auction has started.
```

Comment:
```suggestion
Auctions are conducted in increasing order of block batches, and the next batch's auction can start as soon as the previous batch's auction has ended.
```

Reply:
Actually if auction for block N started, then auction for block N+1 can start